src/error.js

```diff
@@ -550,7 +552,7 @@ export function resetAccumulatedErrorMessagesForTesting() {
  * @visibleForTesting
  */
 export function detectJsEngineFromStack() {
-  /** @constructor */
+  /** @class */
```

Comment:
Why this change?

Reply:
was trying to make linter happy. But type check doesn't like the change. I added `// eslint-disable-line` instead.

cc @rsimha

Reply:
If closure compiler prefers `@constructor` over `@class`, we can specify that preference here. There might be other places where we'd need to change the tag, but it should be a simple find replace, I'd imagine.

https://github.com/ampproject/amphtml/blob/cbf3929bd9e867f38c18c1abeddd1957a7c46e29/.eslintrc#L39-L43

Reply:
Correct, it does prefer `@constructor`: https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#constructor

Reply:
Thanks for the investigation.

I've created #15941. Will have a separate PR to fix this.